pkg/apis/apiserver/organization_REST.go

```diff
@@ -19,25 +19,52 @@ package apiserver
 import (
 	"context"
 	"fmt"
+	"net/http"
 
-	v1 "k8s.io/api/core/v1"
+	authorizationv1 "k8s.io/api/authorization/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/api/meta"
 	"k8s.io/apimachinery/pkg/apis/meta/internalversion"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/apiserver/pkg/endpoints/filters"
 	"k8s.io/apiserver/pkg/registry/generic"
 	"k8s.io/apiserver/pkg/registry/rest"
 	"k8s.io/client-go/dynamic"
+	"k8s.io/klog"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/runtime/inject"
+
+	corev1alpha1 "github.com/kubermatic/bulward/pkg/apis/core/v1alpha1"
+)
+
+const (
+	internalOrganizationResource = "organizations"
+	externalOrganizationResource = "organizations"
+)
+
+var (
+	qualifiedResource = schema.GroupResource{
+		Group:    SchemeGroupVersion.Group,
+		Resource: externalOrganizationResource,
+	}
 )
 
 // +k8s:deepcopy-gen=false
 type OrganizationREST struct {
-	client  client.Client
-	dynamic dynamic.Interface
-	mapper  meta.RESTMapper
-	scheme  *runtime.Scheme
+	client    client.Client
+	dynamicRI dynamic.NamespaceableResourceInterface
```

Comment:
because it's easier passing various options (get/list/create/update options) directly, then converting them to typed client backend. We're writing a proxy, and it's better minimizing conversions as much as possible.

Reply:
for example, Get API interface:

```
	Get(ctx context.Context, key ObjectKey, obj runtime.Object) error
```

you cannot pass minimum RV here, nor objects UID, which are valid `metav1.GetOptions`. Furthermore there are some discrepancies between them as well (for example `metav1.ListOptions` is bigger, for example, `TimeoutSeconds` option, etc.) 

This should be as transparent filtering proxy, thus `client.Client` is wrong abstraction here.